### PR TITLE
[Platform] Unify Linux/FreeBSD modulemaps with gyb

### PR DESCRIPTION
**Pull request by modocache**
#### What's in this pull request?

A great deal of duplication exists across the Linux and FreeBSD modulemaps. An Android modulemap would add even more duplication.

Use gyb to conditionally include certain exports for certain platforms. Most of glibc.modulemap is common across Linux and FreeBSD, and using gyb we can conditionally include the two exports that differ.
#### Resolved bug number: https://github.com/apple/swift/pull/1442#discussion_r58657614

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: stdlib/public/Platform/glibc.modulemap.in.gyb [hunk 1]
```diff
@@ -362,10 +363,12 @@ module SwiftGlibc [system] {
       header "@GLIBC_INCLUDE_PATH@/utmpx.h"
       export *
     }
+% if CMAKE_SDK != "FREEBSD":
```
Comment (gribozavr) [inline stdlib/public/Platform/glibc.modulemap.in.gyb]: Negative tests are fragile and frustrating to read when porting to a new platform.  Could you change these if statements to use comparisons like `CMAKE_SDK in ['LINUX', 'ANDROID']`?

File: stdlib/public/Platform/glibc.modulemap.in.gyb [hunk 2]
```diff
@@ -37,28 +37,29 @@ module SwiftGlibc [system] {
...
 
     module fenv {
       header "@GLIBC_INCLUDE_PATH@/fenv.h"
       export *
     }
-    
+
     // note: supplied by compiler
     // module float {
     //   header "@GLIBC_INCLUDE_PATH@/float.h"
```
Comment (gribozavr) [inline stdlib/public/Platform/glibc.modulemap.in.gyb]: Now that we are using gyb, could we port the `@`-variables to use gyb as well?  Would save some complexity in CMake rules.

**Comment by gribozavr** [comment]
Thank you!  LGTM modulo comments.  Let's get feedback from the CI:
@swift-ci Please test

**Comment by modocache** [comment]
The OS X failure appears to be unrelated.

File: stdlib/public/Platform/glibc.modulemap.in.gyb (hunk 2 above)
Comment (modocache) [inline stdlib/public/Platform/glibc.modulemap.in.gyb]: Ah, very true. Would you mind terribly if I did this in a separate pull request?

File: stdlib/public/Platform/glibc.modulemap.in.gyb (hunk 1 above)
Comment (modocache) [inline stdlib/public/Platform/glibc.modulemap.in.gyb]: Done!

File: stdlib/public/Platform/glibc.modulemap.in.gyb (hunk 2 above)
Comment (gribozavr) [inline stdlib/public/Platform/glibc.modulemap.in.gyb]: No problem!

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by gribozavr** [comment]
CI failures are unrelated.

**Pull request merged**